sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -150,6 +153,14 @@ class DataFrame private[sql](
     }
   }
 
+  private[sql] def resolveToIndex(colName: String): Option[Int] = {
+    val resolver = sqlContext.analyzer.resolver
+    // First remove any user supplied quotes.
+    val unquotedColName = colName.stripPrefix("`").stripSuffix("`")
```

Comment:
do we need to do this? I think for these methods that require column name, user should just pass in an exact column name string, and we don't need to do any extra parsing here, i.e. no resolver, no strip for "`"